android/java/org/chromium/chrome/browser/crypto_wallet/fragments/TxFragment.java

```diff
@@ -407,27 +410,65 @@ private void setupView(View view) {
                 mAssetPrices, mSolanaEstimatedTxFee, mFullTokenList, mNativeAssetsBalances,
                 mBlockchainTokensBalances);
 
+        if (mIsSolanaInstruction) {
+            TextView gasTxTv = view.findViewById(R.id.frag_tx_tv_gas);
+            TextView totalHeading = view.findViewById(R.id.frag_tx_tv_total_heading);
+
+            gasTxTv.setText(R.string.brave_wallet_allow_spend_transaction_fee);
+            totalHeading.setText(R.string.brave_wallet_confirm_transaction_amount_fee);
+            SolanaTxData solanaTxData = TransactionUtils.safeSolData(mTxInfo.txDataUnion);
+            if (solanaTxData != null && solanaTxData.sendOptions != null) {
+                SolanaSendTransactionOptions sendTxOptions = solanaTxData.sendOptions;
+                LinearLayout sendOptionsLl = view.findViewById(R.id.frag_tx_ll_send_options);
+                if (sendTxOptions.maxRetries != null) {
+                    TextView tvLabel = AndroidUtils.makeHeaderTv(requireContext());
+                    TextView tvVal = AndroidUtils.makeSubHeaderTv(requireContext());
+                    tvLabel.setText(R.string.brave_wallet_solana_max_retries);
+                    tvVal.setText(String.valueOf(sendTxOptions.maxRetries.maxRetries));
+                    sendOptionsLl.addView(tvLabel);
+                    sendOptionsLl.addView(tvVal);
+                }
+                if (sendTxOptions.preflightCommitment != null) {
+                    TextView tvLabel = AndroidUtils.makeHeaderTv(requireContext());
+                    TextView tvVal = AndroidUtils.makeSubHeaderTv(requireContext());
+                    tvLabel.setText(R.string.brave_wallet_solana_preflight_commitment);
+                    tvVal.setText(sendTxOptions.preflightCommitment);
+                    sendOptionsLl.addView(tvLabel);
+                    sendOptionsLl.addView(tvVal);
+                }
+                if (sendTxOptions.skipPreflight != null) {
+                    TextView tvLabel = AndroidUtils.makeHeaderTv(requireContext());
+                    TextView tvVal = AndroidUtils.makeSubHeaderTv(requireContext());
```

Comment:
`TextView tvLabel = AndroidUtils.makeHeaderTv(requireContext());
  TextView tvVal = AndroidUtils.makeSubHeaderTv(requireContext());`
could be moved out of 3 ifs as they are same and called once before the conditions

Reply:
They all are three different textviews not one, just using the generic name. We only show these if a corresponding value exists.

Reply:
could they be shown alltogether in some cases?

Reply:
yes, as per the current test case and it's safe to keep the scope limited per case.